packages/2d/src/components/Icon.ts

```diff
@@ -0,0 +1,130 @@
+import {Image, ImageProps} from '../components/Image';
+import {computed, initial, signal} from '../decorators';
+import {
+  DependencyContext,
+  SignalValue,
+  SimpleSignal,
+} from '@motion-canvas/core/lib/signals';
+
+export interface IconProps extends ImageProps {
+  /**
+   * Expects an Icon Identifier.
+   * You can find identifiers on [Icônes](https://icones.js.org/collection/all).
+   * They can look like this:
+   * * mdi:language-typescript
+   * * ph:anchor-simple-bold
+   * * ph:activity-bold
+   */
+  icon: SignalValue<string>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string, with or without # `#bada55`,`141414`
+   *
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   */
+  color?: SignalValue<string>;
+}
+
+/**
+ * An Icon Component that provides an easy access to over 150k icons.
+ * See https://icones.js.org for all available Icons.
+ */
+export class Icon extends Image {
+  /**
+   * Internal cache
+   */
+  private static iconContentsPool: Record<string, HTMLImageElement> = {};
+
+  /**
+   * Expects an Icon Identifier.
+   * You can find identifiers on [Icônes](https://icones.js.org).
+   * They can look like this:
+   * * mdi:language-typescript
+   * * ph:anchor-simple-bold
+   * * ph:activity-bold
+   */
+  @signal()
+  public declare icon: SimpleSignal<string, this>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string, with or without # `#bada55`,`141414`
+   *
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   */
+  @initial('white')
+  @signal()
+  public declare color: SimpleSignal<string, this>;
+
+  public constructor(props: IconProps) {
+    super(props);
+  }
+
+  /**
+   * This creates the URL that will be used as the Image source
+   * @returns Address to Iconify API for the requested Icon.
+   */
+  @computed()
+  protected svgUrl(): string {
+    const iconPathSegment = this.icon().replace(':', '/');
+
+    // iconify API is documented here: https://docs.iconify.design/api/svg.html#color
+    const url = `https://api.iconify.design/${iconPathSegment}.svg?color=${Icon.normalizeColorInput(
+      this.color(),
+    )}`;
+    return url;
+  }
+
+  private static normalizeColorInput(colorString: string) {
```

Comment:
Looking at this again, we probably can condense this into two steps:
1. Check if there's a hash, and if there is make sure that it's a valid hex number of 6 or 8 digits.
    * If there's not, warn the user that they used an invalid hex using the logger and don't bother updating the value.
2. Send the result to the `encodeURIComponent` function. If there's a `#`, it'll get converted. If not, it stays as the input string.

Reply:
Also just found a simpler way to check if its a valid color.
Assuming that every CSS color is valid you could use:

```ts
function isColor(color: string){
  const style = new Option().style;
  style .color = color;
  return style .color === color;
}
```

Reply:
>  warn the user that they...

How would I access the console inside the UI? printing to `console.error` does not seem to print the Error there.

Reply:
You can use the built-in logger as follows:
```ts
useLogger().error('Some error here');
```
There is also a PR (#227) on the way for further references.

Reply:
> Assuming that every CSS color is valid you could use:

Just checked, and yup :) The API just passes through whatever you provide it as the color to the SVG Path's `fill` property. 

e.g. `https://api.iconify.design/mdi/access-point.svg?color=testtest` 
sets the fill to testtest
![image](https://user-images.githubusercontent.com/27686785/218221855-9e286c06-5efe-48cb-89b4-489c7f30f119.png)

looks like the most elegant solution in that case :+1:

Reply:
> Should work with Hex Colors

On Firefox it returns the color in the rgb() syntax unfortunately. 
![image](https://user-images.githubusercontent.com/27686785/218225993-c1934c27-b5de-418f-bd31-866af7e65d18.png)

Reply:
With a color signal, you could just do `this.color().hex()` to retrieve the hex value.